src/actions/pkg.cr

```diff
@@ -1,45 +1,142 @@
-module DppmRestApi::Actions::Pkg
-  extend self
-  # List built packages
-  relative_get nil do |context|
-    if context.current_user? && Actions.has_access? context, Access::Read
-      # TODO: list all built packages
-      next context
+module DppmRestApi::Actions
+  module Pkg
+    extend self
+    ALL_PKGS = ""
+    ONE_PKG  = "/:id"
+    include RouteHelpers
+
+    def build_config_response(package, builder)
+      builder.field package.package do
+        builder.object do
+          package.each_config_key do |key|
+            builder.field key, package.get_config key
+          end
+        end
+      end
     end
-    raise Unauthorized.new context
-  end
-  # Clean unused built packages
-  relative_delete "/clean" do |context|
-    if context.current_user? && Actions.has_access? context, Access::Delete
-      # TODO: delete all unused built packages
-      next context
+
+    def clean_unused_packages(context : HTTP::Server::Context)
+      if result = DppmRestApi.prefix.clean_unused_packages(confirmation: false) { }
+        return {data: result} if result.any?
+        bug = InternalServerError.new context, "received empty set from Prefix#clean_unused_packages; please report this strange bug"
+        error = NoPkgsToClean.new context
+        error.cause = bug
+        raise error
+      end
+      raise NoPkgsToClean.new context
     end
-    raise Unauthorized.new context
-  end
-  # Query information about a given package
-  relative_get "/:id/query" do |context|
-    if context.current_user? && Actions.has_access? context, Access::Read
-      # TODO: Query information about the given package
-      next context
+
+    # List built packages
+    relative_get nil do |context|
+      if Actions.has_access? context, Access::Read
+        pkgs = [] of String
+        begin
+          DppmRestApi.prefix.each_pkg { |pkg| pkgs << pkg.package }
```

Comment:
It shouls be a list of hashes, with at least the package name and the version. I think it's a sane default.
In the future (or now if you prefer), this elements will be configurable through parameters.
An user may wants also descriptions, documentation link, license etc.

Reply:
> this elements will be configurable through parameters. An user may wants also descriptions, documentation link, license etc.

A user will not be using the API directly. If the client wants to query that information about a package, they could use the *query* route.

Reply:
Why? This means when the Web client will list all applications, it will also have to do multiple requests?
We will have duplicate packages with this current implementation when we have two versions of the same package.

Reply:
Fair enough.